src/compiler_pipeline.rs

```diff
@@ -196,6 +199,7 @@ impl<E, Extra> Compileable<Extra> for TypecheckValue<E>
                compiler: &mut Compiler,
                thread: &Thread,
                filename: &str,
+               expr_str: &str,
```

Comment:
Wondering if we could be passing in a `Source` at this stage. Ditto with the other methods.